lib/ruby/lib/alces/stack/hosts/run.rb

```diff
@@ -50,33 +51,30 @@ def run!
           when @dry_run_flag
             lambda = dry_run
           when @add_flag
-            lambda = -> (json) {add(json)}
-          else
-            raise "Could not modify hosts, see 'metal hosts -h'"
+            lambda = -> (template_parameters) {add(template_parameters)} 
           end
 
-          Alces::Stack::Iterator.new(@nodegroup, lambda, @json) if !lambda.nil?
+          Alces::Stack::Iterator.run(@nodegroup, lambda, @template_parameters) if !lambda.nil?
+          raise "Could not modify hosts! No command included (e.g. --add).\nSee 'metal hosts -h'" if lambda.nil?
         end
 
         def dry_run
           case
           when @add_flag
-            lambda = -> (json) {puts_template(json)}
-          else
-            raise "Could not modify hosts, see 'metal hosts -h'"
+            lambda = -> (template_parameters) {puts_template(template_parameters)}
           end
           return lambda
         end
 
-        def add(json)
+        def add(template_parameters)
           append_file = "/etc/hosts"
-          json = "" if !json
-          Alces::Stack::Templater::JSON_Templater.append(@template, append_file, json, @template_parameters)
+          @json = "" if !@json
```

Comment:
A good pattern for this is: `@json ||= ""` Ref: https://github.com/bbatsov/ruby-style-guide#double-pipe-for-uninit